zetta_utils/builder/parser.py

```diff
@@ -2,13 +2,17 @@
 from __future__ import annotations
 
 import copy
-from typing import Any, Callable, TypeVar, Union
+from typing import Any, Callable, List, Optional, TypeVar, Union
 
 from typeguard import typechecked
 
 from zetta_utils.partial import ComparablePartial
 
+from ..typing import IntVec3D, Vec3D
+
 REGISTRY: dict = {}
+REGISTRY_CAST_VEC3D: dict = {}
+REGISTRY_CAST_INTVEC3D: dict = {}
```

Comment:
I'd prefer the `REGISTRY: dict[str, RegistryEntry]` approach, where
```python
@attrs.mutable
class RegistryEntry:
        fn: Callable
        name: str
        cast_vec3d: ...
```

Reply:
Sounds good, though I guess how this is implemented depends on the answer to the next question.